File: 프로그래머스/3/12979. 기지국 설치/기지국 설치.js
function solution(n, stations, w) {
    var answer = 0;
    let lastIdx = 1
    for(var station of stations){
        let res = Math.ceil(((station-w)-lastIdx)/(2*w+1));
        answer+=res;
        lastIdx=station+w+1;
    }
    if(lastIdx<n){
    return answer+Math.ceil((n-lastIdx)/(2*w+1));
    }
    if(lastIdx>n) return answer;
    return answer+1;
}

File: 프로그래머스/3/12979. 기지국 설치/기지국 설치.java
import java.util.*;
class Solution {
    public int solution(int n, int[] stations, int w) {
        int answer = 0;
        int startIdx = 1;
        int endIdx = 0;
        for(int s : stations){
            endIdx = s-w-1;
            double d = (double)(endIdx-startIdx+1)/(2*w+1);
            if(startIdx<=endIdx) answer += (int)Math.ceil(d);
            startIdx=s+w+1;
        }
        if(startIdx<=n) answer+=Math.ceil((double)(n-startIdx+1)/(2*w+1));
        return answer;
    }
}